quic/s2n-quic/src/tests/zero_length_cid_client_connection_migration.rs

```diff
@@ -0,0 +1,234 @@
+            let sockets = vec![&socket, &migrated_socket];
+            for active_socket in sockets {
+                let local_addr = active_socket.local_addr().unwrap();
+                match active_socket.try_recv_from() {
+                    Ok(Some((from, _ecn, payload))) => {
+                        // Quiche conn.recv requires a mutable payload array
+                        let mut payload_copy = payload.clone();
+
+                        // Feed received data from IO Socket to Quiche
+                        let _read = match client_conn.recv(
+                            &mut payload_copy,
+                            quiche::RecvInfo {
+                                from,
+                                to: active_socket.local_addr().unwrap(),
+                            },
+                        ) {
+                            Ok(v) => v,
+                            Err(quiche::Error::Done) => 0,
+                            Err(e) => {
+                                panic!("quiche client receive error: {e:?}");
+                            }
+                        };
+                    }
+                    Ok(None) => {}
+                    Err(e) => {
+                        panic!("quiche client socket recv error: {e:?}");
+                    }
+                }
+
+                for peer_addr in client_conn.paths_iter(local_addr) {
+                    loop {
+                        let (write, send_info) = match client_conn.send_on_path(
+                            &mut out,
+                            Some(local_addr),
+                            Some(peer_addr),
+                        ) {
+                            Ok(v) => v,
+                            Err(quiche::Error::Done) => {
+                                break;
+                            }
+                            Err(e) => {
+                                panic!("quiche client send error: {e:?}")
+                            }
+                        };
+
+                        active_socket
+                            .send_to(send_info.to, NotEct, out[..write].to_vec())
+                            .unwrap();
+                    }
+
+                    // Send application data using the migrated address
+                    // This can only be done once the connection migration is completed
+                    if local_addr == migrated_socket.local_addr().unwrap()
+                        && client_conn
+                            .is_path_validated(local_addr, peer_addr)
+                            .unwrap()
+                        && !req_sent
+                    {
+                        client_conn
+                            .stream_send(QUICHE_STREAM_ID, application_data.as_bytes(), true)
+                            .unwrap();
+                        req_sent = true;
+                    }
+                }
+
+                for stream_id in client_conn.readable() {
```

Comment:
is it possible for the test to pass even if the client connection never becomes readable, but is closed for some other reason?

Reply:
Is it possible the client could initiate a close on its own due to an idle timeout? Maybe we could set a boolean after verifying that the stream data matches and check that before breaking on line 223